### ISSUE TITLE
BitmapFont in Box2D world [Enhancement]

### ISSUE THREAD

**Issue by obigu** [libgdx/libgdx#3827]
Box2D requires real world unit based coordinate system to work properly (1 unit = 1 meter). The logical choice is to have a small viewport such as 32 * 24 or even smaller depending on the game logic to avoid transformations.

Because BitmapFont uses pixels and not world units it needs to be scaled to a value between 0.01 to 0.1 to be displayed in the right size in such small viewports (otherwise it looks huge). Unfortunately BitmapFont doesn't display properly when scaled with values under 0.1, glyphs start disappearing and its behaviour is messed up.

Some other people have faced this problem before:
http://gamedev.stackexchange.com/questions/73688/why-is-my-text-is-too-large-even-when-scaled-to-05f-in-libgdx
http://www.java-gaming.org/index.php?topic=30929.0

A common suggested workaround is to use a different camera but that is not a solution if the text is part of the game (not the UI, think for instance of a 2048 type of game using Box2D or any other pyshics based puzzle game with text).

The alternative "solution" is to define a larger viewport and then transform every single displayable Box2D object dimensions before rendering (typycally defining a PxelsToMeters constant) which is far from ideal.

The information is there but it gets lost in the transformations.

Could BitmapFont be enhanced to work with small viewports? Would it be possible to somehow support providing the transformation matrix to the World coordinates to avoid extreme scaling or, alternatively, improve it to display correctly when using these scaling values?

**Comment by Darkyenus** [comment]
> Unfortunately BitmapFont doesn't display properly when scaled with values under 0.1, glyphs start disappearing and its behaviour is messed up.

BitmapFont works well in these scales, but you have to disable rounding to integers. You'll lose the benefits of rounding to integers, but that is usable only in 1:1 unit:pixel ratio anyway and any workaround to make it work in other ratios would be complicated, so probably won't happen.

TLDR: `BitmapFont#setUseIntegerPositions(false)`, not a libGDX bug

**Comment by obigu** [comment]
Awesome, missed that one, thanks!
